search/views.py

```diff
@@ -44,11 +47,19 @@ def query(request):
                 path = k
                 k = k.split('/')
                 ans.append({'path': path, 'dirs': k, 'files': filtered_v})
-
+                d = k[len(k)-2] + '/'
```

Comment:
This should be: `d = k[len(k)-3] + '/'` 

The reason being we will go one level down to find the recommended search results. For instance: if we search `fibonacci search`, then it is likely that we will be interested in other search algorithms. 

In the current state, the recommended results seem to be random to the end user. 

We need further refinements such that results like "search/test/` are avoided, if going one level down does not give the required number of results, then go one further level down and other such considerations.

Reply:
Do I need to add `k[len(k)-3] + '\'` to only those which does not give the required number of search results? 
If not, I am afraid to tell that `d = k[len(k)-3] + '\'` works only to those codes whose directory list length is more than two. For instance, if we take Poisson sample the directory list length is 2.

Reply:
Ideally, the directory list length is 3 but discrepancies may exist. 

For `poisson sample` keyword, the directory list is `mathematical_algorithms/src/poisson_sample/`, so it will return results from the mathematical section. You have to take care that if the condition fails, we are able to give a less-refined recommended result.